build(docs-infra): update Angular framework/Material/CLI and Lighthouse

gkalpak opened the pull request:
See individual commits for details.

##
Regarding the payload size increases, they are mostly attributed to Angular Material:

| Change | Size | Increase |
|--|--:|--:|
| Before this PR | 448461 B ||
| After framework update | 448554 B | +93 B |
| After Material update | 449292 B | +738 B |
| After CLI update | 449310 B | +18 B |

mary-poppins commented:
You can preview c2a57b3 at https://pr40994-c2a57b3.ngbuilds.io/.

mary-poppins commented:
You can preview 7a3e6e8 at https://pr40994-7a3e6e8.ngbuilds.io/.

mary-poppins commented:
You can preview 1d36564 at https://pr40994-1d36564.ngbuilds.io/.

jessicajaniuk reviewed (approved):
LGTM 🍪 

reviewed-for: size-tracking

AndrewKushnir reviewed (approved):
Reviewed-for: size-tracking

Splaktar commented on aio/scripts/audit-web-app.js:
```diff
@@ -35,7 +35,6 @@ const puppeteer = require('puppeteer');
 // Constants
 const AUDIT_CATEGORIES = ['accessibility', 'best-practices', 'performance', 'pwa', 'seo'];
 const LIGHTHOUSE_FLAGS = {logLevel: process.env.CI ? 'error' : 'info'};  // Be less verbose on CI.
-const SKIPPED_HTTPS_AUDITS = ['redirects-http', 'uses-http2'];
```
Can you provide any details on why this was removed? Are these audits no longer relevant for this newer Lighthouse version? or is there an assumption here that no one will run these tests locally?

gkalpak replied:
I assume you missed the info in the commit message (or do you mean more details than that?):

> In this version, we no longer need our custom logic to skip certain HTTPS related audits on localhost, since Lighthouse will treat it as a secure context (similar to how browsers do).
> See also GoogleChrome/lighthouse#11766.

Basically, we no longer need to exclude these on localhost (which does not use HTTPS) to get the same score as on production (where we do use HTTPS).

(BTW, we do run PWA score tests against localhost [on CI][1], so we still care about localhost scores :wink:)

[1]: https://github.com/angular/angular/blob/be8893fd1dbe1620c880bd067c02f168d5809d62/.circleci/config.yml#L380-L381

Splaktar replied:
Good to know! Thank you. And yes, I ignored your first line about reading the individual commit messages 🤦🏼

gkalpak committed:
build(docs-infra): update Lighthouse to version 7.2.0

In this version, we no longer need our custom logic to skip certain
HTTPS related audits on localhost, since Lighthouse will treat it as a
secure context (similar to how browsers do).
See also GoogleChrome/lighthouse#11766.

gkalpak committed:
build(docs-infra): update Angular framework, Material and CLI to latest methods

This commit updates the Angular framework, Angular CDK/Material and
Angular CLI to latest stable versions (11.2.3, 11.2.2 and 11.2.2
respectively).

This update also fixes a Lighthouse audit fail due to
`@angular/core@11.0.0` being identified as vulnerable to XSS:
https://snyk.io/vuln/SNYK-JS-ANGULARCORE-1070902

Regarding the payload size increases, they are mostly attributed to
Angular Material:
- Before this commit:     448461 B
- After framework update: 448554 B ( +93 B)
- After Material update:  449292 B (+738 B)
- After CLI update:       449310 B ( +18 B)

gkalpak commented:
FYI: Rebased on master to resolve merge conflict.

mary-poppins commented:
You can preview fe59b11 at https://pr40994-fe59b11.ngbuilds.io/.

mary-poppins commented:
You can preview 7c98c7d at https://pr40994-7c98c7d.ngbuilds.io/.

zarend commented:
Hi @gkalpak we merged this into `master` because this branch has merge conflict with the `11.2.x` branch. If this needs to go into patch, because make another PR for this with `target: patch` and can merge into `11.2.x` branch with conflict.
